packages/core/useFetch/index.ts

```diff
@@ -0,0 +1,161 @@
+import { Ref, ref, unref, watch, isRef, computed, ComputedRef } from 'vue-demi'
+import { Fn, MaybeRef } from '@vueuse/shared'
+
+interface UseFetchReturn {
+  /**
+   * Indicates if the fetch request has finished
+   */
+  isFinished: Ref<boolean>
+
+  /**
+   * The statusCode of the HTTP fetch response
+   */
+  status: Ref<number | null>
+
+  /**
+   * The raw response of the fetch response
+   */
+  response: Ref<Response | null>
+
+  /**
+   * Any fetch errors that may have occured
+   */
+  error: Ref<any>
+
+  /**
+   * The fetch response body, may either be JSON or text
+   */
+  data: Ref<object | string | null>
+
+  /**
+   * Indicates if the request is currently being fetched.
+   */
+  isFetching: Ref<boolean>
+
+  /**
+   * Indicates if the fetch request is able to be aborted
+   */
+  canAbort: ComputedRef<boolean>
+
+  /**
+   * Abort the fetch request
+   */
+  abort: Fn
+
+  /**
+   * Manually call the fetch
+   */
+  execute: Fn
+}
+
+interface UseFetchOptions {
+  /**
+   * Will automatically run fetch when `useFetch` is used
+   *
+   * @default true
+   */
+  immediate?: boolean
+
+  /**
+   * Will automatically refetch when the URL is changed if the url is a ref
+   *
+   * @default false
+   */
+  refetch?: MaybeRef<boolean>
+}
+
+export function useFetch(url: MaybeRef<string>): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, useFetchOptions: UseFetchOptions): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, options: RequestInit, useFetchOptions?: UseFetchOptions): UseFetchReturn
+
+export function useFetch(url: MaybeRef<string>, ...args: any[]): UseFetchReturn {
+  const supportsAbort = typeof AbortController === 'function'
+
+  let fetchOptions: RequestInit = {}
+  let options: UseFetchOptions = { immediate: true, refetch: false }
+
+  if (args.length > 0) {
+    if ('immediate' in args[0] || 'refetch' in args[0])
+      options = { ...options, ...args[0] }
+    else
+      fetchOptions = args[0]
+  }
+
+  if (args.length > 1) {
+    if ('immediate' in args[1] || 'refetch' in args[1])
+      options = { ...options, ...args[1] }
+  }
+
+  const isFinished = ref(false)
+  const isFetching = ref(false)
+  const status = ref<number | null>(null)
+  const response = ref<Response | null>(null)
+  const error = ref<any>(null)
+  const data = ref<string | object | null>(null)
+
+  const canAbort = computed(() => {
+    return supportsAbort && isFetching.value
+  })
+
+  let controller: AbortController | null
```

Comment:
```suggestion
  let controller: AbortController | undefiend
```
So it matches with the initial value

Reply:
Oh yeah fixed that in my last commit